### PR TITLE
Add with_sub method for ServiceAccountKey

### PR DESCRIPTION
**Pull request by e-oz**
required to read email from user accounts (not service account).

### CODE REVIEW

**Comment by Byron** [comment]
@dermesser Somehow the travis notification didn't get through, even though it is happy. What do you think about this PR?

File: src/service_account.rs [hunk 1]
```diff
@@ -192,11 +193,23 @@ impl<'a, C> ServiceAccountAccess<C>
             client: client,
             key: key,
             cache: MemoryStorage::default(),
+            sub: None,
+        }
+    }
+
+    pub fn with_sub(key: ServiceAccountKey, client: C, sub: Option<String>) -> ServiceAccountAccess<C> {
```
Comment (dermesser) [inline src/service_account.rs]: how about making the `sub` argument a non-Option `String`? After all, if no sub is needed, `new()` should be used anyway.

**Review by dermesser — CHANGES_REQUESTED** [review]
I tested it with the service account example, it at least doesn't break it :) LGTM aside from the one comment.

File: src/service_account.rs (hunk 1 above)
Comment (e-oz) [reply src/service_account.rs]: you are right, agree

**Pull request merged**